test.bash

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+# Authority key autosigning testcases
+# SPDX-License-Identifier: BSD-2-Clause
+# SPDX-FileCopyrightText: 2019 Michał Górny
+# SPDX-FileCopyrightText: 2023-2024 Robin Johnson
+die() {
+	local rc=$1
+	shift
+	echo "$*" 1>&2
+	[[ -n ${AUTOSIGN_TMPDIR} ]] && echo "Testing tmpdir is ${AUTOSIGN_TMPDIR}" 1>&2
+	exit "$rc"
+}
+
+if [[ -z ${LOCAL_KEYSERVER} && -z ${AUTOSIGN_NO_SEND_KEYS} ]]; then
+	die 1 "Please set LOCAL_KEYSERVER to local keyserver address for pushing, or AUTOSIGN_NO_SEND_KEYS=1 to disable pushing."
+fi
+
+TEST_RC=0
+
+# For testing, keep a single tmpdir, and iterate phases.
+AUTOSIGN_TMPDIR=$(mktemp -d /tmp/autosign-test.XXXXXX) || die 2 "Failed to create testing directory"
+GNUPGHOME=${AUTOSIGN_TMPDIR}/gnupghome
+export AUTOSIGN_TMPDIR GNUPGHOME
+echo "Testing tmpdir is ${AUTOSIGN_TMPDIR}"
+
+umask 077 && mkdir -p "${GNUPGHOME}"
+[ -n "${AUTOSIGN_NO_SEND_KEYS}" ] || echo "keyserver ${LOCAL_KEYSERVER}" >"${GNUPGHOME}"/gpg.conf
```

Comment:
```suggestion
[[ -n ${AUTOSIGN_NO_SEND_KEYS} ]] || echo "keyserver ${LOCAL_KEYSERVER}" >"${GNUPGHOME}"/gpg.conf
```

Again, inconsistent style all over the place. Am I expected to point every single one of them?

Reply:
What format-checker do you recommend then since neither shfmt-3.1.2 nor shellcheck-0.10.0 are catching these, and I missed two inconsistencies only.

This is resolved again